.ci_support/osx_64_numpy1.19python3.7.____cpython.yaml

```diff
@@ -9,7 +9,7 @@ channel_targets:
 cxx_compiler:
 - clangxx
 cxx_compiler_version:
-- '13'
```

Comment:
@traversaro @wolfv  I think we have the same issue as in https://github.com/conda-forge/pinocchio-feedstock/pull/76 with eigenpy
As far I understand, these lines are the origin, and I think this is related to Boost, but not sure 100%.

Reply:
I think we can add some lines in the `conda_build_config.yaml` to ask to still use llvm 13 (I need to check how) but this is a workaround.

Reply:
Yes, fine. But I think there is an issue in the upstream OSX packages. Don't you think so?

Reply:
Yes, that could be an issue on LLVM/clang. To investigate more, I enabled pinocchio C++ tests in https://github.com/conda-forge/pinocchio-feedstock/pull/76 .

Reply:
This maybe means that we need to rerender boost-python, no?

Reply:
I remembered some issues concerning some aspects of compilations flags used by conda when compiling boost. This was a long time ago.

Reply:
> This maybe means that we need to rerender boost-python, no?

The boost is currently pinned to 1.74 while the latest boost is 1.80, so we would need to rerender the v1.74 branch : https://github.com/conda-forge/boost-feedstock/tree/v1.74.x . However, this would not resolve runtime issues as the conda solver would not be aware of this constraint, but for sure it is better than nothing.

Reply:
> this would not resolve runtime issues as the conda solver would not be aware of this constraint,

I don't really understand here. What do you mean by 'not aware'?

Reply:
My bad. From what we hypothesized, to avoid the "Illegal Instructions", you need that both eigenpy and boost-python libraries are compiled with the same clang major version. Let's imagine that we have the _0 builds compiled with clang 13 and _1 builds with clang 14. If we just install eigenpy, the latest versions (_1) will be installed, so there is no problem. However if we install them in more complex environements, there is no costraint at the conda level that tells the solver that for both packages only the _1 build can be installed, so it may happen that boost-python _1 is installed with eigenpy _0 and viceversa.

Reply:
OK. Clear.
Do you have a conda forcing procedure to avoid that?

Reply:
But, here, what I don't understand, is that this appends for eigenpy directly, which should use the same compiler than boost-python, no?

Reply:
> Do you have a conda forcing procedure to avoid that?

Perhaps there exists one, but at the moment I do not know if something like that exists.

> But, here, what I don't understand, is that this appends for eigenpy directly, which should use the same compiler than boost-python, no?

Not sure I am following you here. In this PR, `eigenpy` is using clang 14, while I guess that boost-python package is using 13 or earlier package.

Reply:
OK. I understand the issue now. Could it be possible then to update boost-python, or should we wait later?
But for me, such an issue should not happen. clang 13 and 14 should be compliant. I guess there is an issue here.

Reply:
Yes, also in my experience clang 13 and clang 14 compiled products are normally ABI-compatible. However, note that we did not validate this hypothesis, so it is possible that the illegal operation appears even if boost-python is compiled with clang 14 .

Reply:
Thanks a lot @traversaro for the deep investigation.